File: EmojibleAndroid/app/src/main/java/com/dji/emojibleandroid/dataSources/ClassesDataSource.kt
package com.dji.emojibleandroid.dataSources

import com.google.firebase.firestore.ListenerRegistration
import com.google.firebase.firestore.ktx.firestore
import com.google.firebase.ktx.Firebase

class ClassesDataSource {
    var database = Firebase.firestore
    var snapShotListener: ListenerRegistration? = null

    private object HOLDER {
        val INSTANCE = ClassesDataSource()
    }

    companion object {
        val instance: ClassesDataSource by lazy { HOLDER.INSTANCE }
    }
}

File: EmojibleIOS/EmojibleIOS/DataSources/ClassesDataSource.swift
//
//  ClassesDataSource.swift
//  EmojibleIOS
//
//  Created by Haldun Balim on 12.11.2020.
//

import Foundation
import Firebase

class ClassesDataSource{
    let database = Firestore.firestore()
    var snapshotListener:ListenerRegistration!
    
    private init(){
    }
    private static let instance = ClassesDataSource()
    public static func getInstance() -> ClassesDataSource{
        return .instance
    }
    
    public func signUpToClass(classCode:String, classPassword:String, completion: @escaping (String?) -> Void){
        
    }
    
}
